imap_processing/idex/idex_l2c.py

```diff
@@ -0,0 +1,236 @@
+"""
+Perform IDEX L2c Processing.
+
+Examples
+--------
+.. code-block:: python
+    from imap_processing.idex.idex_l1a import PacketParser
+    from imap_processing.idex.idex_l1b import idex_l1b
+    from imap_processing.idex.idex_l1b import idex_l2a
+    from imap_processing.idex.idex_l1b import idex_l2b
+
+    l0_file = "imap_processing/tests/idex/imap_idex_l0_sci_20231214_v001.pkts"
+    l1a_data = PacketParser(l0_file)
+    l1b_data = idex_l1b(l1a_data)
+    l1a_data = idex_l2a(l1b_data)
+    l2b_data = idex_l2b(l2a_data)
+    write_cdf(l2b_data)
+"""
+
+import logging
+
+import astropy_healpix.healpy as hp
+import numpy as np
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.ena_maps.ena_maps import SkyTilingType
+from imap_processing.ena_maps.utils.coordinates import CoordNames
+from imap_processing.ena_maps.utils.spatial_utils import AzElSkyGrid
+from imap_processing.idex.idex_constants import (
+    IDEX_HEALPIX_NESTED,
+    IDEX_HEALPIX_NSIDE,
+    IDEX_POINTING_REFERENCE_FRAME,
+    IDEX_SPACING_DEG,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def idex_l2c(l2b_dataset: xr.Dataset) -> list[xr.Dataset]:
+    """
+    Will process IDEX l2b data to create l2c data products.
+
+    Parameters
+    ----------
+    l2b_dataset : xarray.Dataset
+        IDEX L2b dataset.
+
+    Returns
+    -------
+    l2b_dataset : list[xarray.Dataset]
+        The``xarray`` dataset containing the science data and supporting metadata.
+    """
+    logger.info(
+        f"Running IDEX L2C processing on datasets: "
+        f"{l2b_dataset.attrs['Logical_source']}"
+    )
+    # For ISTP, epoch should be the center of the time bin.
+    epoch = xr.DataArray(
+        [np.mean(l2b_dataset["epoch"].data[[0, -1]]).astype(np.int64)],
+        name="epoch",
+        dims=["epoch"],
+    )
+    l2c_healpix_dataset = idex_healpix_pset(l2b_dataset, epoch)
+    l2c_rectangular_dataset = idex_rectangular_pset(l2b_dataset, epoch)
```

Comment:
Change these method names to `idex_healpix_map`? i.e. I think you are skipping the intermediate pset creation now and binning directly in the map frame.